consensus/istanbul/backend/api.go

```diff
@@ -474,6 +477,100 @@ func (api *APIExtension) GetBlockWithConsensusInfoByHash(blockHash common.Hash)
 	return api.makeRPCBlockOutput(block, cInfo, block.Transactions(), receipts), nil
 }
 
+func (api *APIExtension) GetBlsInfos(number rpc.BlockNumber) (map[string]interface{}, error) {
+	kip113Addr, err := api.GetActiveAddressFromRegistry(system.Kip113Name, number)
+	if err != nil {
+		return nil, err
+	}
+
+	var bn *big.Int
+	if number == rpc.LatestBlockNumber || number == rpc.PendingBlockNumber {
+		bn = big.NewInt(api.chain.CurrentBlock().Number().Int64())
+	} else {
+		bn = big.NewInt(number.Int64())
+	}
+
+	backend := backends.NewBlockchainContractBackend(api.chain, nil, nil)
+	if infos, err := system.ReadKip113All(backend, kip113Addr, bn); err != nil {
+		return nil, err
+	} else {
+		blsInfos := make(map[string]interface{})
+		// make publicKey, Pop to hexadecimal string
+		for addr, info := range infos {
+			blsInfos[addr.Hex()] = map[string]interface{}{
+				"publicKey": hex.EncodeToString(info.PublicKey),
+				"pop":       hex.EncodeToString(info.Pop),
+				"verifyErr": info.VerifyErr,
+			}
+		}
+		return blsInfos, nil
+	}
+}
+
+func (api *APIExtension) GetAllRecordsFromRegistry(name string, number rpc.BlockNumber) ([]interface{}, error) {
+	var bn *big.Int
+	if number == rpc.LatestBlockNumber || number == rpc.PendingBlockNumber {
+		bn = big.NewInt(api.chain.CurrentBlock().Number().Int64())
+	} else {
+		bn = big.NewInt(number.Int64())
+	}
+
+	if api.chain.Config().IsRandaoForkBlock(bn) && name == system.Kip113Name {
```

Comment:
If `IsRandaoForkBlock`, then we can just return `config.RandaoRegistry.Records[name]`.
In KIP-149, the RandaoRegistry can contain any records other than KIP113.

Reply:
Thanks. Applied at https://github.com/klaytn/klaytn/pull/2111/commits/4d846e9cb11f4f4d3d98a9fc96403732bdecf816

Reply:
As @ian0371's below review, we don't need this branch for api (of course, still needed in `getAllCached`). I remove this and rollback to original `geAllCached` at https://github.com/klaytn/klaytn/pull/2111/commits/d61e68dcfa4644208bc424de2620bdb7638eea33